Scripts/Ebon Hold: Into the Realm of Shadows

Sorikoff opened the pull request:
**Changes proposed:** 
- Removed obsolete C++ script.
- Created separate SAI scripts for `Dark Rider of Acherus`, `Acherus Deathcharger`, and `Salanar the Horseman`.
- Changed spell click spells to correct ones.
- Slightly reorganized code.

**Target branch(es):** 3.3.5

~**Issues addressed:**~

**Tests performed:** Works as intended.

**Known issues and TODO list:**:
- [ ] after removing player from vehicle, camera is getting stuck in place.
- [ ] personal spawn support in core.

Sorikoff committed:
Scripts/Ebon Hold: Into the Realm of Shadows

Sorikoff committed:
Scripts/Ebon Hold: Into the Realm of Shadows

Sorikoff committed:
Update used spells

Sorikoff committed:
Update used spells

illfated commented:
Cosmetic/codestyle: missing NewLine at the end of [9999_99_99_99_world.sql](https://github.com/TrinityCore/TrinityCore/pull/25617/files#diff-41736fd024e478d43ea196b36e4c48290630e19da51c8bb8cff3b705199330e6) .

illfated commented on sql/updates/world/3.3.5/9999_99_99_99_world.sql:
```diff
@@ -0,0 +1,38 @@
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=13 AND `SourceEntry`=52349;
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
+(18,28782,46598,0,0,31,0,3,0,0,0,0,0,'','Allow spellclick for NPC'),
+(18,28782,52349,0,0,9,0,12687,0,0,0,0,0,'','Required quest 12687 active for spellclick'),
```
Suggestion for improvement of the `Comment` field:
```suggestion
(18,28782,52349,0,0,9,0,12687,0,0,0,0,0,'',"Requires quest 'Into the Realm of Shadows' active for spellclick"),
```

Sorikoff committed:
Further fixes

Sorikoff committed:
Further fixes

Sorikoff commented on src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:
```diff
@@ -865,87 +824,21 @@ class spell_stable_master_repo : public AuraScript
-        caster->SetFaction(FACTION_FRIENDLY);
-
-        caster->CastSpell(caster, SPELL_CALL_DARK_RIDER, true);
+        GetHitUnit()->ExitVehicle();
```
After removing aura `52349` camera is getting stuck in place (character moves though). Could someone please help with this issue? Currently have no idea.

illfated replied:
Maybe I have misunderstood something in these changes, but while viewing the code in split screen, it looks like aura 52349 is not handled by any enum or reference. Could that be part of the issue?

Sorikoff replied:
It's not used in C++ code (only applied in SQL).

illfated replied:
Fair enough, I misunderstood how it is handled.

Riztazz commented:
This quest is supposed to use a spell to summon the final NPC, 52363, on top of that, that spawn is a personal spawn (i don't think 3.3.5 has that implemented but pretty sure master does) and from what we've gathered with @Traesh a year ago, it's a personal spawn.
https://github.com/TrinityCore/TrinityCore/blob/c250a858b91a11736ee0a64c2f120b1d036b5203/src/server/game/DataStores/DBCEnums.h#L1346

Sorikoff commented:
Thank you for this note! As far as I see, this indeed is not implement in 3.3.5. https://github.com/TrinityCore/TrinityCore/commit/b7bb5e6a98119512702519adecf559ecedc20743 probably needs to be ported.

illfated commented on sql/updates/world/3.3.5/9999_99_99_99_world.sql:
```diff
@@ -0,0 +1,54 @@
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=13 AND `SourceEntry`=52349;
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
+(18,28782,46598,0,0,31,0,3,0,0,0,0,0,'','Allow spellclick for NPC'),
+(18,28782,52349,0,0,9,0,12687,0,0,0,0,0,'','Requires quest 'Into the Realm of Shadows' active for spellclick'),
```
To use a double set of quotes in a comment, you need to use 1 set of double quotes and 1 set or single quotes, or use an escape character (backslash) for the inner single quotes. My suggestion:
```suggestion
(18,28782,52349,0,0,9,0,12687,0,0,0,0,0,'',"Requires quest 'Into the Realm of Shadows' active for spellclick"),
```
(CircleCI codestyle_and_sql https://app.circleci.com/pipelines/github/TrinityCore/TrinityCore/2992/workflows/70193d78-f2a1-467d-963b-33db6e28192e/jobs/12641)

Ovahlord commented:
You forgot to remove the scriptname for npc_salanar_the_horseman and you did not add the SAI for him yet so you cannot enter the shadow realm, Jimbo.

However, I went on testing this stuff and so far the Horse neither the rider do reset property so they are stuck in combat. Another thing that I have noticed is that Salanar does not walk towards you when delivering the horse which he should do when you call him. About the vehicle camera issue, I was not able to reproduce it after using.

```
        if (Vehicle* vehicle = GetHitUnit()->GetVehicle())
            vehicle->RemoveAllPassengers();
```

Sorikoff commented:
Thank you for feedback. Removed ScriptName. Not sure about SAI for him though as you enter shadow realm automatically. At least it works that way for me.

Vehicle camera issue is still present for me. Will check reset.

Sorikoff committed:
Remove ScriptName

Sorikoff committed:
Remove ScriptName

Sorikoff committed:
SQL

Update sql/updates/world/3.3.5/9999_99_99_99_world.sql

Co-authored-by: Trond B. Krokli <38162891+illfated@users.noreply.github.com>

Sorikoff committed:
SQL

Update sql/updates/world/3.3.5/9999_99_99_99_world.sql

Co-authored-by: Trond B. Krokli <38162891+illfated@users.noreply.github.com>

illfated commented on sql/updates/world/3.3.5/9999_99_99_99_world.sql:
```diff
@@ -0,0 +1,56 @@
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=13 AND `SourceEntry`=52349;
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
+(18,28782,46598,0,0,31,0,3,0,0,0,0,0,'','Allow spellclick for NPC'),
+(18,28782,52349,0,0,9,0,12687,0,0,0,0,0,'','Requires quest 'Into the Realm of Shadows' active for spellclick'),
```
Looks like you removed the solution to use only 1 pair each of double quotes (" ") and single quotes (' ').

Maybe it is easier for you to remember if you use the escape character "backslash" \ for the inner quote pair, which is actually meant to transfer into the DB comment text field.

```suggestion
(18,28782,52349,0,0,9,0,12687,0,0,0,0,0,'','Requires quest \'Into the Realm of Shadows\' active for spellclick'),
```

Sorikoff replied:
Yeah, I just used force push, and forgot to pull PR changes.

jackpoz committed:
Core/Player: Adjust fall damage log

Move fall damage log to "entities.player.falldamage" category and include the usual debug info

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 57)

Closes #25653

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 58)

Closes #25659

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 59)

Closes #25661

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 60)

Closes #25665

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 61)

Closes #25673

offl committed:
DB/Battleground/Alterac: Update stats

offl committed:
DB/Instance/Arcatraz: More adjustments

offl committed:
DB/Instance/Eye of Eternity: Verify & update spawn positions & stats

offl committed:
DB/Creature: Remove wrongly added despawn on death action from Flawless Arcane Elemental

Closes #25668

illfated committed:
DB/Quest: Add missing CompletionText & RewardText for Pilgrim's Bounty quests

Closes #25671

offl committed:
DB/Creature: Add CREATURE_FLAG_EXTRA_NO_COMBAT to jewelcrafting stone statues & root them

Closes #20406

offl committed:
DB/Creature: Add script to Thornling

offl committed:
DB/SAI: Add or update few scripts

Ovahlord committed:
DB/Quest: Add correct emotes to some Scarlet Enclave quests

offl committed:
DB/Creature: Properly root few creatures & update their scripts

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 62)

Closes #25676

offl committed:
DB/Creature: Update Silithid Creeper Egg & Silithid Grub

offl committed:
DB/Creature: Update Apothecary Thedra & script Halis Dawnstrider

offl committed:
DB/Creature: Spawn missing A Mysterious Voice & add texts

offl committed:
DB/Onslaught Harbor: Update Captain Welsington, Captain Hartford & Onslaught Darkweaver

offl committed:
DB/Quest: Update 'Putting Olakin Back Together Again' (13220)

offl committed:
DB/Quest: Rework 'Through the Eye' (13121)

offl committed:
DB: Update subzone Kul'galar Keep

offl committed:
DB/Gameobject: Fix position of Goldthorn

Closes #25681

offl committed:
DB/Quest: Rework 'Delivering the Message' (10406)

Closes #25666

offl committed:
DB/Loot: Create common skinning loot tables for some silithids

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 63)

Closes #25685

offl committed:
DB/Loot: Create common skinning loot tables for more creatures

offl committed:
DB/Quest: Add PrevQuestID to 'Flash Bomb Recipe' (1559)

Closes #25686

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 64)

Closes #25687

offl committed:
DB/Creature: Make Drakkari Rhino immune to polymorph

offl committed:
DB/Creature: Remove skinning loot from some unskinnable creatures

offl committed:
DB/Quest: Rework 'Plundering the Plunderers' (2381)

schell244 committed:
DB/Loot: Captain's Key & Pirate's Footlocker

Closes #25689

jackpoz committed:
Improve multithreading of mmaps_generator (#25625)

* Build/Misc: Add a few *San CMake flags

Add the following flags for the related tools:
- MSAN for Memory Sanitizer
- UBSAN for Undefined Behavior Sanitizer
- TSAN for Thread Sanitizer

* Remove unused parameter

* Fix UBSan reported issue

* Disable G3D buffer pools when using Thread Sanitizer as it has its custom locking mechanisms

* Code cleanup

* Move threads from maps to tiles

* Move tile building logic to TileBuilder class

* Fix memory leak in TileBuilder

* Fix build

* Store TileBuilder as raw pointer for now, it will be changed later on to use modern C++ constructs

* Fix crash on shutdown

* Revert pvs-studio change

* Fix generating 1 single tile not closing the program

offl committed:
DB/Quest: Update 'Metzen the Reindeer'

offl committed:
DB/Quest: Rework 'Revenge for the Vargul' (13059)

offl committed:
DB/Quest: Rework 'The Sum is Greater than the Parts' (13043)

offl committed:
DB/Creature: Bethod Feigr & Valhalas Vargul

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 65)

Closes #25691

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 66)

Closes #25693

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 67)

Closes #25694

offl committed:
DB/Creature: Properly apply CREATURE_FLAG_EXTRA_NO_SKILLGAIN

Closes #25674

jackpoz committed:
Core/Movement: Add a debug log when a player falls below the map

offl committed:
DB/Creature: Make Metzen the Reindeer's event not repeatable

Ref f45d2d6597a9c8cb97ae5f20dbd52be91c3eae71

offl committed:
DB/Instance/Sethekk Halls: Adjustments for Anzu & fix positions of some creatures

offl committed:
DB/Instance/Shadow Labyrinth: Adjustments for Grandmaster Vorpil & fix stats

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 68)

Closes #25695

Treeston committed:
Game/ChatCommands: Make sure ChatHandler::extractPlayerTarget populates the strtok buffer for non-nullptr returns. Closes #25692, supersedes #25698.

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 69)

Closes #25700

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 70)

Closes #25703

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 71)

Closes #25713

Commit:
TDB 335.20121 - 2020/12/15

offl committed:
DB/Quest: Conditions for 'Winter's Presents' & 'New Year Celebrations!'

Closes #25733

Closes #25734

offl committed:
DB/Spell: Isle of Quel'Danas & Sunwell Plateau area spells

offl committed:
DB/Instance/Sunwell Plateau: Overhaul part 1

offl committed:
DB/Instance/Sunwell Plateau: Overhaul part 2

offl committed:
DB/Creature: Apprentice Osterkilgr

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 72)

Closes #25725

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 73)

Closes #25728

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 74)

Closes #25737

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 75)

Closes #25739

offl committed:
DB/Creature: Root Ironwork Cannon & fix model

offl committed:
DB/Instance/Sunwell Plateau: Overhaul part 3

offl committed:
DB/Quest: Update 'The Seer's Crystal' & 'Corki's Gone Missing Again!'

offl committed:
DB/Creature: Update stats & auras of some creatures

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 76)

Closes #25747

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 77)

Closes #25751

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 78)

Closes #25753

offl committed:
DB/Quest: Update 'Ebon Blade Prisoners' (12982)

offl committed:
DB/Quest: Rework 'The Honored Ancestors' (11605)

offl committed:
DB/Quest: Update 'Cycle of Life' (12454)

offl committed:
DB/Spell: Add effects to some serverside spells

offl committed:
DB/Creature: Wyrmrest Protector & Warden of the Chamber

offl committed:
DB/Creature: Use serverside spells to activate taxi at Wyrmrest Temple

offl committed:
DB/Spell: Add effects to all summoning spells from Violet Hold

offl committed:
Core/Scripts: Add missing spells to Violet Hold

gonzo1247 committed:
Core/Vehicles: Add missing caster guid check

Merges #25773

(cherry picked from commit e26122dc54b5c5a356a97a842718168dab97a0aa)

jackpoz committed:
CI/GitHub: Apply branch label to PRs when they are opened

Apply branch label to PRs when they are opened instead of having a cron job that runs every 5 minutes. This uses the "new" github action type "pull_request_target"

(cherry picked from commit 89cfced1b568e8807ab71f7da0e3ffc5cdc10ce7)

# Conflicts:
#	.github/workflows/pr-labeler.yml

jackpoz committed:
Core/Spells: Implemented personal summons (#19231) (#25765)

* Core/Spells: Implemented personal summons (#19231)

* By default determined by summon property flag SUMMON_PROP_FLAG_PERSONAL_SPAWN

Closes #18254

(cherry picked from commit b7bb5e6a98119512702519adecf559ecedc20743)

# Conflicts:
#	src/server/game/Combat/ThreatManager.cpp
#	src/server/game/DataStores/DBCEnums.h
#	src/server/game/Entities/Creature/TemporarySummon.cpp
#	src/server/game/Entities/Creature/TemporarySummon.h
#	src/server/game/Entities/GameObject/GameObject.cpp
#	src/server/game/Entities/Object/Object.cpp
#	src/server/game/Entities/Object/Object.h
#	src/server/game/Maps/Map.h
#	src/server/game/Spells/SpellEffects.cpp

* Build fix

* Implement feedback

* Fix parameters passed in wrong order

Jildor committed:
Core/Visibility: Split BG/Arena visibility settings (#25690)

Ovahlord committed:
Core/SAI: implement new smart actions to change AI Animkits and to enable/disable hover movement (#25623)

* Core/SAI: implement new smart actions to change AI Animkits and to enable/disable hover movement for units

SMART_ACTION_SET_AI_ANIM_KIT (parameter 1: animKitId)
SMART_ACTION_SET_HOVER (parameter1: enable/disable)

* Update SmartScriptMgr.cpp

* Update SmartScript.cpp

* removed SMART_ACTION_SET_AI_ANIM_KIT since it's reserved for master

jackpoz commented:
> * personal spawn support in core.

this has been added with https://github.com/TrinityCore/TrinityCore/commit/3b2c878dd067f49ee48ddfc27c62eb4e1fe619d9 , please update this PR to include whatever required personal spawns

Ovahlord committed:
Core/Units: do not send power update packets when gaining power via energize spell effects to match retail behavior (#25599)

This fixes possible clientside visual issues where power is being added twice until the next update_object packet is being sent

offl committed:
DB/Creature: Add conditions to Steam Tonk's spells & misc fixes

Ref ed8bf7f0e2e67ca1eb65ffa05518fe29df526642 Updates #9373

offl committed:
DB/Creature: Replace unit flags stun & immune with auras

Commit:
DB/Quest: Add missing CompletionText to 2 quests

Closes #25752

Commit:
DB/Locales: Add missing esES & esMX quest locales (Part 79)

Closes #25766

offl committed:
DB/Spell: Add effects to more spells from Violet Hold

Turned out almost all spells from BM were copied to create Violet Hold. Not only serverside spells and not only their effects but also attributes and icons. The scripts were also copied, possibly unchanged

offl committed:
Core/Scripts: Add missing spells to Black Morass

jackpoz committed:
Scripts/Commands: Include map min height in ".gps" command

jackpoz committed:
Core/Maps: Use the same logic to calculate grid coords to avoid mismatches

Use the same logic of (CENTER_GRID_ID - x / SIZE_OF_GRIDS) in GridMap::getMinHeight() as Map::GetGrid() to avoid coord grid mismatches caused by floating point precision, i.e. getting the wrong grid.
Replace defines with constexpr in grids to make debugging easier.

Closes #25652

jackpoz committed:
COre/Misc: Remove whitespace

jackpoz committed:
Core/Maps: Partially revert 0c69316714d0558e32462f5401197fe1e803ae75

The development will continue in a PR

jackpoz committed:
Core/Maps: Partially revert 0c69316714d0558e32462f5401197fe1e803ae75

Revert define to constexpr as this also broke some calculations

Shauren committed:
Core/SmartAI: Reserve a param for SMART_ACTION_SOUND master branch specific functionality, for compatibility reasons

Shauren committed:
Core/Vehicles: Treat entering other vehicles mounted on main vehicle as swapping seats

(cherry picked from commit 110564e757fefbfbddfab876a96ca9eb5aa7dd7b)

jackpoz committed:
Core/Maps: Attempt to fix picking the wrong min height (#25771)

Close #25652

jackpoz committed:
Scripts/ICC: Fix teleporter not removing players from Gunship

A more generic solution might be better, maybe depending on spell flags

Ref #25652

Sorikoff committed:
Merge branch '3.3.5-12687' of https://github.com/Sorikoff/TrinityCore into 3.3.5-12687

Faq commented:
@Sorikoff You will open new PR?

illfated commented:
Looks like `https://github.com/Sorikoff` does not contain any current repository. Will have to re-fork TC first to start working on it from that account.

jackpoz commented:
a new PR with just these commits should be re-opened:
1. https://github.com/TrinityCore/TrinityCore/pull/25617/commits/9895ab9104113af633f535e4957c30537720dfdd
2. https://github.com/TrinityCore/TrinityCore/pull/25617/commits/7b160b3bd2bfe6c38c26820d27a3375e980cdc53
3. https://github.com/TrinityCore/TrinityCore/pull/25617/commits/73d1b35adad543d3ecfffced0f46bbbcf9ebb055
4. https://github.com/TrinityCore/TrinityCore/pull/25617/commits/3f3d45cd3f429d8d25da344bde0cc71d08d8312c
5. https://github.com/TrinityCore/TrinityCore/pull/25617/commits/68f525559c1f60f0df59cef9425210773e9789ad

illfated commented:
I presume it will be useful for me to open a new PR just to consolidate those commits into 1, in case anyone else wants to work on adding any missing C++ coding or SAI. (Also useful to have it open and available for review, suggestions etc.)

jackpoz commented:
@illfated if possible please keep the 5 commits split in the PR

illfated commented:
> @illfated if possible please keep the 5 commits split in the PR

Very well. Thanks for the correction. Glad I have not started doing anything major, only mixed together the SQL part, so it is easy enough to use the commits. I presume any updates (improvements) from my side will have to be separate commits as well.

offl commented:
> in case anyone else wants to work on adding any missing C++ coding or SAI. (Also useful to have it open and available for review, suggestions etc.)

This PR was opened for 2 months for review and suggestions
Merging PR mostly requires changes from OP and not from reviewers

illfated commented:
Sure, that is why I offered to open a new PR for the changes in here (with a note saying that @Sorikoff could have the credit, or even remake this original PR).

offl commented:
I mean if you are not going to update and test it, very likely it will be too not merged

illfated commented:
Fair point. I thought as much, but also thought that it would be useful to have the material in an open PR. Oh well. Point taken.
